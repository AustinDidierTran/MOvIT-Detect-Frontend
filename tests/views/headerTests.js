import React from 'react';
import TestUtils from 'react-dom/test-utils';
import PropTypes from 'prop-types';
import Header from '../../src/views/header';

describe('Header Tests', () => {
  it('should have proptypes', function () {
    // Actual value
    const actualValue = Header.WrappedComponent.propTypes;

    // Expected value
    const expectedValue = {
      language: PropTypes.string.isRequired,
      profile: PropTypes.string,
      changeLanguage: PropTypes.func,
      changeProfile: PropTypes.func,
      changeToken: PropTypes.func,
    };

    // Test
    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });
});
