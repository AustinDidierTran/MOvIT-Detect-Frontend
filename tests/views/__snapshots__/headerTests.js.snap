// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests should match the snapshot 1`] = `
<div
  className="top headerNav"
>
  <nav
    className="navbar navbar-expand-lg"
  >
    <Link
      className="navbar-brand title"
      replace={false}
      to="/home"
    >
      MOvIT+
    </Link>
    <button
      aria-controls="navbarContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler custom-toggler"
      data-target="#navbarContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarContent"
    >
      <ul
        className="nav navbar-nav mr-auto"
      >
        <li
          className="nav-item px-3 mt-1"
          data-target=".navbar-collapse.show"
          data-toggle="collapse"
        >
          <Link
            className="navLink"
            replace={false}
            to="/goals"
          >
            goals.FR
          </Link>
        </li>
        <li
          className="nav-item px-3 mt-1"
          data-target=".navbar-collapse.show"
          data-toggle="collapse"
        >
          <Link
            className="navLink"
            replace={false}
            to="/results"
          >
            graphics.FR
          </Link>
        </li>
      </ul>
      <li
        className="nav navbar-nav pl-1"
      >
        <a
          className="small-font navLink"
          onClick={[Function]}
        >
          EN
        </a>
      </li>
      <span>
        <li
          className="nav navbar-nav pl-1"
          data-target=".navbar-collapse.show"
          data-toggle="collapse"
        >
          <Link
            className="navLink px-2"
            replace={false}
            to="/settings"
          >
            <i
              className="fa fa-cog"
            />
          </Link>
        </li>
        <li
          className="nav navbar-nav pl-1"
        >
          <Link
            className="navLink"
            onClick={[Function]}
            replace={false}
            to="/home"
          >
            welcome.logout.FR
            <i
              className="fa fa-sign-out pl-2"
            />
          </Link>
        </li>
      </span>
    </div>
  </nav>
</div>
`;
